server/app.py

```diff
@@ -36,18 +36,23 @@ def create_app(csrf: CSRFProtect, db: SQLAlchemy):
     db.init_app(app)
     csrf.init_app(app)
     jwt = JWTManager(app)
-    web.setup(app)      # FIXME deprecated package
-    execution.web.setup.setup(app)
+
+    # -- Endpoint connection
+    execution.web_api.setup.setup(app)
+    scenario.web_api.setup.setup(app)
+    execution.api.setup.setup(app)
+
     CORS(app, resources={r"/*": {"origins": "*"}})
 
-    # define run request blocker
+    # -- Pre Request Handler
     @app.before_request
     def return_hold_if_not_running():
+        """ Blocks all run-request on an execution which is still in status pending. """
```

Comment:
I think, we should think about the semantics of this function. Currently, it does not block anything. However, if an execution is _not running_, we simply return `204` (which, additionally, means `No Content`, but we return a message - thus not following the HTTP semantics). 

If this function returns nothing, we just execute the regular request handlers.

Reply:
This function detects if an execution is not set to status running. If so, the function simply returns an empty message. Therefore it blocks any state changing requests on the api. afaik this method is tested and proven. However we can argue about the location of the request handler. We can relocate it to the registered api endpoints only and exclude the web-api.